ledger-core/v2/virtual/integration/convlog/logger.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package convlog
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"runtime"
+	"strings"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/log/logfmt"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
+)
+
+type MachineLogger struct {
+}
+
+func (MachineLogger) LogMachineInternal(data smachine.SlotMachineData, msg string) {
+	fmt.Printf("[MACHINE][LOG] %s[%3d]: %03d @ %03d: internal %s%s\n", data.StepNo.MachineID(), data.CycleNo,
+		data.StepNo.SlotID(), data.StepNo.StepNo(), msg, formatErrorStack(data.Error))
+}
+
+func (MachineLogger) LogMachineCritical(data smachine.SlotMachineData, msg string) {
+	fmt.Printf("[MACHINE][ERR] %s[%3d]: %03d @ %03d: internal %s%s\n", data.StepNo.MachineID(), data.CycleNo,
```

Comment:
Maybe it should use logger instead of plain fmt?

Reply:
this logger is purely for local debug and made to be **readable**, not for logs

Reply:
added a build tag